### PR TITLE
Add Code of Conduct 

### PR DESCRIPTION
**Pull request by tommy351**
## What does it do?

Netlify recently updated their [policy](https://www.netlify.com/legal/open-source-policy/?utm_campaign=Netlify+%3C3+Open+Source+email&utm_content=Netlify+open+source+plan&utm_medium=email_action&utm_source=customer.io). It is required to have a Code of Conduct in the repository.

## How to test

```sh
git clone -b BRANCH https://github.com/USER/hexo.git
cd hexo
npm install
npm test
```

## Pull request tasks

- [ ] Add test cases for the changes.
- [ ] Passed the CI test.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at hi@hexo.io. All
```
Comment (SukkaW) [inline CODE_OF_CONDUCT.md]: Maybe a dedicated email address could be used, like `abuse@hexo.io` or `report@hexo.io`?

Comment (curbengh) [reply CODE_OF_CONDUCT.md]: @NoahDragon mentioned about mail forward, not sure the current status.

**Comment by tomap** [comment]
Quick question: you used the v1.4. but there is a v2 available. Why not use it?
https://www.contributor-covenant.org/version/2/0/code_of_conduct

MD: https://www.contributor-covenant.org/version/2/0/code_of_conduct.md

**Comment by tommy351** [comment]
@tomap This pull request is created by the template on GitHub. I didn't know there's a newer version of CoC.

**Comment by SukkaW** [comment]
@YoshinoriN Agree. We should add the link of Code of Conduct at the beginning the README, just as Node.js do in their README.

**Comment by tommy351** [comment]
I updated the code of conduct and readme in https://github.com/hexojs/hexo/pull/3962/commits/d851678852627220c71db3111b0c0c3473a7184f

Here I list some items to do before we merged:
- [ ] Make sure `report@hexo.io` is available.
- [ ] Make sure reports sent to the email address can be handled.
- [ ] Add the link of CoC to the [contribution guide](https://hexo.io/docs/contributing) as well.

**Comment by curbengh** [comment]
> Quick question: you used the v1.4. but there is a v2 available. Why not use it?
> https://www.contributor-covenant.org/version/2/0/code_of_conduct

I thought @tommy351 just adapt from [Babel](https://github.com/babel/babel/blob/master/CODE_OF_CONDUCT.md) project (who still using v1.4). I find v2 a bit harsh.

**Comment by tcrowe** [comment]
For people searching in the future, if you want to know how hexo org deals with abuse of contributors please view my analysis here: https://github.com/tcrowe/code-communists/blob/master/projects/hexo/index.md

I was a contributor who was taking a break. My plan to come back and work on hexo more but I've removed myself from the hexo org over this issue.

**Comment by SukkaW** [comment]
@NoahDragon Please help setting up `report@hexo.io` email address at mailgun. Every "owner" of hexojs organization should be able to receive related complaint email.

**Comment by NoahDragon** [comment]
@SukkaW That's a good idea. I have created the email address report[at]hexo.io.
@hexojs/core By default, the email only forwards to owners, but core members could also join voluntarily. Just shot an email to the above address if you would like to be involved.

**Comment by SukkaW** [comment]
@tommy351 Sinc the email address `report at hexo.io` is now up, I believe it is good to merge this PR then. I will add the link to Code of Conduct to contributing guideline later.

**Comment by tommy351** [comment]
圖片

Quick update: We are re-qualified for the open source plan now. 🎉

**Pull request merged**